service/authenticate.py

```diff
@@ -0,0 +1,113 @@
+import bcrypt
+from db.database import retrieve_filter,users,tasks
+import jwt
+import os
+from dotenv import load_dotenv
+from bson.objectid import ObjectId
+from bson.errors import InvalidId
+from fastapi import Depends,status,HTTPException,Header
+from fastapi.responses import JSONResponse
+from fastapi.encoders import jsonable_encoder
+from fastapi import Path
+import json
+load_dotenv()
+
+seceret_key = os.getenv("SECERET_KEY")
+
+def hash_password(password: str) -> bytes:
+    salt = bcrypt.gensalt()
+    hashed_password = bcrypt.hashpw(password.encode('utf-8'), salt)
+    return hashed_password
+
+def gen_jwt(data: dict) -> str:
+    token = jwt.encode(data, seceret_key, algorithm="HS256")
+    return token
+
+async def get_user(token : str = Header(...)) -> dict:
+    try:
+        user = jwt.decode(token,seceret_key,algorithms="HS256")
+        user = jsonable_encoder(user)
+        if await retrieve_filter(users,{"_id" : ObjectId(user["user_id"])}):
+            return user
+        else:
+            raise ValueError("")
+    except jwt.InvalidTokenError:
+        return JSONResponse(
+            content={
+                "error" : "invalid Token"
+            },
+            status_code=status.HTTP_401_UNAUTHORIZED
+        )
+
+async def authenticated_login(username: str, rqpassword: str) -> str:
+    user = await retrieve_filter(users,{"username" : username})
+    if user:
+        password = user["password"]
+        user.pop("password") 
+        if bcrypt.checkpw(rqpassword.encode('utf-8'), password): 
+            token = gen_jwt({"username": user["username"], "user_id": user["_id"],"role" : user["role"]} )  
+            return token
+    return None
+
+class RoleBaseAccessControlWithTask:
```

Comment:
This file name is authenticate.py, so you should move this class to authorize.py or something like that. Do not messy function of each class or function into 1 file